src/app/infrastructure/pawnMoves.ts

```diff
@@ -0,0 +1,219 @@
+import {
+    Cord,
+    IChessBoard,
+    CordWithMoveType,
+    MoveType,
+    ChessBoardRepresentation,
+    PieceType,
+    Piece,
+    Side,
+} from '../domain/basicChessTypes';
+//import { IChessEngine } from '../domain/IChessEngine';
+
+enum MoveDirection {
+    LEFT = -1,
+    RIGHT = 1,
+    UP = 1,
+    DOWN = -1,
+}
+
+function isOutsideBoard(cord: Cord['x'] | Cord['y']): boolean {
+    if (cord > 7 || cord < 0) {
+        return true;
+    }
+    return false;
+}
+
+function isBlockedTile(tile: Piece | null): boolean {
+    if (tile != null && tile != undefined) {
+        return true;
+    }
+    return false;
+}
+
+function isOppositePawn(currentPawn: Piece | null, pieceToComparison: Piece | null): boolean {
+    if (currentPawn === null || pieceToComparison == null) {
+        return false;
+    }
+    if (pieceToComparison.figType == PieceType.Pawn && pieceToComparison.side != currentPawn.side) {
+        return true;
+    }
+    return false;
```

Comment:
Skoro domyślnie i tak zwracasz fałsz, nie lepiej zapisać poniższej funkcji w taki sposób?
```ts
return pieceToComparison.figType == PieceType.Pawn && pieceToComparison.side != currentPawn.side ? true : false
```

Reply:
No tak trzy returny, zapis jest dobry, musiałem tylko uwzględnić możliwość null poprzez dodanie znaku zapytania pieceToComparison?.figType oraz currentPawn?.side i błąd typescripta został wyeliminowany

Reply:
To ja może jeszcze bardziej skrócę zapis:
```ts
return pieceToComparison.figType == PieceType.Pawn && pieceToComparison.side != currentPawn.side;
```